File: Demo/InternationalDemo/Objective-C/BUDemo/App/Example/view/BUDNativeView.m
//
//  BUADVADemo
//
//  Created by bytedance in 2022.
//  Copyright © 2022 bytedance. All rights reserved.

#import <PAGAdSDK/PAGLNativeAdRelatedView.h>
#import <PAGAdSDK/PAGLMaterialMeta.h>
#import <PAGAdSDK/PAGMediaView.h>
#import <PAGAdSDK/PAGLNativeAd.h>

#import "BUDNativeView.h"

@interface BUDNativeView ()

@property (nonatomic, strong) PAGLNativeAdRelatedView *relatedView;
@property (nonatomic, strong) UILabel *titleLabel;
@property (nonatomic, strong) UILabel *detailLabel;

@end

@implementation BUDNativeView

- (instancetype)initWithFrame:(CGRect)frame {
    
    if (self = [super initWithFrame:frame]) {
        
        self.backgroundColor = UIColor.lightGrayColor;
        
        self.relatedView = PAGLNativeAdRelatedView.new;
        [self addSubview:self.relatedView.mediaView];
        [self addSubview:self.relatedView.dislikeButton];
        [self addSubview:self.relatedView.logoADImageView];
        
        self.titleLabel = UILabel.new;
        self.titleLabel.textAlignment = NSTextAlignmentCenter;
        self.titleLabel.textColor = UIColor.blackColor;
        [self addSubview:self.titleLabel];
        
        self.detailLabel = UILabel.new;
        self.detailLabel.textAlignment = NSTextAlignmentCenter;
        self.detailLabel.textColor = UIColor.blackColor;
        [self addSubview:self.detailLabel];
        
        [self addSubview:self.relatedView.adChoicesView];
    }
    return self;
}

- (void)layoutSubviews {
    [super layoutSubviews];
        
    self.relatedView.mediaView.frame = CGRectMake(0, 50, CGRectGetWidth(self.bounds) , CGRectGetHeight(self.bounds) - 100);
    self.relatedView.dislikeButton.frame = CGRectMake(0, 0, 44, 44);
    CGSize logoSize = CGSizeMake(20, 10);
    CGSize adChoiceSize = CGSizeMake(14, 14);
    self.relatedView.logoADImageView.frame = CGRectMake(0, CGRectGetHeight(self.bounds) - logoSize.height, logoSize.width, logoSize.height);
    self.relatedView.adChoicesView.frame = CGRectMake(CGRectGetWidth(self.bounds) - adChoiceSize.width, CGRectGetHeight(self.bounds) - adChoiceSize.height, adChoiceSize.width, adChoiceSize.height);
    self.titleLabel.frame = CGRectMake(0, 0, CGRectGetWidth(self.bounds), 20);
    self.detailLabel.frame = CGRectMake(0, CGRectGetHeight(self.bounds) - 40, CGRectGetWidth(self.bounds), 40);
}

- (void)refreshWithNativeAd:(PAGLNativeAd *)nativeAd {
    self.titleLabel.text = nativeAd.data.AdTitle;
    self.detailLabel.text = nativeAd.data.AdDescription;
    [self.relatedView refreshWithNativeAd:nativeAd];
}

@end

File: Demo/InternationalDemo/Swift/BUADVADemo_Swift/ExampleApp/SubPage/BUDNativeView.swift
//
//  BUDNativeView.swift
//  BUADVADemo_Swift
//
//  Created by Willie on 2022/5/11.
//

import UIKit

class BUDNativeView: UIView {

    private lazy var relatedView: PAGLNativeAdRelatedView = {
        let view = PAGLNativeAdRelatedView()
        return view
    }()

    private lazy var titleLabel: UILabel = {
        let label = UILabel()
        label.textAlignment = .center
        label.textColor = .black
        return label
    }()
    
    private lazy var detailLabel: UILabel = {
        let label = UILabel()
        label.textAlignment = .center
        label.textColor = .black
        return label
    }()
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        
        backgroundColor = .lightGray
        
        addSubview(relatedView.mediaView)
        addSubview(relatedView.dislikeButton)
        addSubview(relatedView.logoADImageView)
        addSubview(titleLabel)
        addSubview(detailLabel)
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func layoutSubviews() {
        super.layoutSubviews()
        
        relatedView.mediaView.frame = self.bounds
        relatedView.dislikeButton.frame = CGRect(x: 0, y: 0, width: 44, height: 44)
        let logoSize: CGSize  = CGSize(width: 20, height: 10)
        relatedView.logoADImageView.frame = CGRect(x:0,
                                                   y:self.bounds.maxY - logoSize.height,
                                                   width:logoSize.width,
                                                   height:logoSize.height)
        titleLabel.frame = CGRect(x:0, y:0, width:self.bounds.maxX, height:20)
        detailLabel.frame = CGRect(x:0, y:self.bounds.maxY - 40, width:self.bounds.maxX, height:40)
    }
    
    func refresh(with nativeAd: PAGLNativeAd) {
        titleLabel.text = nativeAd.data.adTitle
        detailLabel.text = nativeAd.data.adDescription
        relatedView.refresh(with :nativeAd)
    }
}
